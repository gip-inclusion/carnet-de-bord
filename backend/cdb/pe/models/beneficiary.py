from typing import Optional

from pydantic import BaseModel


class Beneficiary(BaseModel):
    identifiant: Optional[str] = None
