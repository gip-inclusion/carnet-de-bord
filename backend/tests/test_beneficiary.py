from datetime import date

from asyncpg.connection import Connection
from syrupy.data import Snapshot

from cdb.api.db.crud.beneficiary import (
    get_beneficiaries_without_referent,
    get_beneficiary_from_personal_information,
)
from cdb.api.db.crud.professional_project import (
    find_professional_project_for_beneficiary,
)
from cdb.api.db.models.beneficiary import (
    Beneficiary,
    BeneficiaryImport,
    BeneficiaryWithAdminStructureEmail,
)
from cdb.cdb_csv import pe
from cdb.cdb_csv.models.csv_row import PrincipalCsvRow


async def test_get_beneficiary_with_professional_projects(
    pe_principal_csv_series, db_connection
):

    # Get the first row
    _, series = next(pe_principal_csv_series.iterrows())
    csv_row: PrincipalCsvRow = await pe.map_principal_row(series)

    beneficiary: Beneficiary | None = await get_beneficiary_from_personal_information(
        db_connection,
        firstname=csv_row.prenom,
        lastname=csv_row.nom,
        birth_date=csv_row.date_naissance,
    )

    assert beneficiary is not None

    if beneficiary:
        assert beneficiary.lastname == "Tifour"
        assert beneficiary.firstname == "Sophie"
        assert beneficiary.date_of_birth == date(1982, 2, 1)
        assert beneficiary.right_ass is False
        assert beneficiary.right_are is False
        assert beneficiary.right_bonus is False
        assert beneficiary.right_rsa == "rsa_droit_ouvert_et_suspendu"
        assert beneficiary.notebook is not None
        assert len(beneficiary.notebook.professional_projects) == 2

        assert (
            await find_professional_project_for_beneficiary(
                beneficiary, csv_row.rome_1, csv_row.appelation_rome_1
            )
        ) is not None


async def test_get_beneficiary_without_professional_projects(
    pe_principal_csv_series, db_connection
):

    for idx, series in pe_principal_csv_series.iterrows():
        # Get the second element
        if idx == 1:

            csv_row: PrincipalCsvRow = await pe.map_principal_row(series)

            beneficiary: Beneficiary | None = (
                await get_beneficiary_from_personal_information(
                    db_connection,
                    firstname=csv_row.prenom,
                    lastname=csv_row.nom,
                    birth_date=csv_row.date_naissance,
                )
            )

            assert beneficiary is not None

            if beneficiary:
                assert beneficiary.lastname == "Dorsey"
                assert beneficiary.firstname == "Hendrix"
                assert beneficiary.date_of_birth == date(1976, 12, 18)
                assert beneficiary.right_ass is False
                assert beneficiary.right_are is False
                assert beneficiary.right_bonus is False
                assert beneficiary.right_rsa is None
                assert beneficiary.notebook is not None
                assert len(beneficiary.notebook.professional_projects) == 0

                assert (
                    await find_professional_project_for_beneficiary(
                        beneficiary, csv_row.rome_1, csv_row.appelation_rome_1
                    )
                ) is None


async def test_get_beneficiary_without_notebook(pe_principal_csv_series, db_connection):

    for idx, series in pe_principal_csv_series.iterrows():
        # Get the third element
        if idx == 2:

            csv_row: PrincipalCsvRow = await pe.map_principal_row(series)

            beneficiary: Beneficiary | None = (
                await get_beneficiary_from_personal_information(
                    db_connection,
                    firstname=csv_row.prenom,
                    lastname=csv_row.nom,
                    birth_date=csv_row.date_naissance,
                )
            )

            assert beneficiary is not None

            if beneficiary:
                assert beneficiary.notebook is None


async def test_get_beneficiary_with_unknown_rome_code(
    caplog, pe_principal_csv_series, db_connection
):

    for idx, series in pe_principal_csv_series.iterrows():
        # Get the fourth element
        if idx == 3:

            csv_row: PrincipalCsvRow = await pe.map_principal_row(series)

            beneficiary: Beneficiary | None = (
                await get_beneficiary_from_personal_information(
                    db_connection,
                    firstname=csv_row.prenom,
                    lastname=csv_row.nom,
                    birth_date=csv_row.date_naissance,
                )
            )

            assert beneficiary is not None

            if beneficiary:
                assert beneficiary.lastname == "Skinner"
                assert beneficiary.firstname == "Edwina"
                assert beneficiary.date_of_birth == date(1973, 5, 14)
                assert beneficiary.right_ass is True
                assert beneficiary.right_are is True
                assert beneficiary.right_bonus is False
                assert beneficiary.right_rsa is None
                assert beneficiary.notebook is not None
                assert len(beneficiary.notebook.professional_projects) == 0

                assert (
                    await find_professional_project_for_beneficiary(
                        beneficiary, csv_row.rome_1, csv_row.appelation_rome_1
                    )
                ) is None

                await pe.check_and_insert_professional_project(
                    db_connection,
                    beneficiary.notebook,
                    csv_row.rome_1,
                    csv_row.appelation_rome_1,
                )

                assert "Rome code not found" in caplog.text


def test_beneficiary_model_beneficiary_filled_keys(
    beneficiary_import_alain_die: BeneficiaryImport,
):
    editable_keys = beneficiary_import_alain_die.get_beneficiary_editable_keys()
    editable_keys.sort()
    assert editable_keys == [
        "address1",
        "address2",
        "city",
        "email",
        "mobile_number",
        "place_of_birth",
        "postal_code",
        "right_ass",
        "right_rsa",
    ]


def test_beneficiary_model_notebook_filled_keys(
    beneficiary_import_alain_die: BeneficiaryImport,
):
    editable_keys = beneficiary_import_alain_die.get_notebook_editable_keys()
    editable_keys.sort()
    assert editable_keys == [
        "work_situation",
    ]


def test_beneficiary_get_values_for_keys(
    beneficiary_import_alain_die: BeneficiaryImport,
):
    beneficiary_import_alain_die.right_are = True
    beneficiary_import_alain_die.right_ass = True
    beneficiary_import_alain_die.right_bonus = True
    values = beneficiary_import_alain_die.get_values_for_keys(
        [
            "right_are",
            "right_ass",
            "right_bonus",
            "right_rsa",
            "work_situation",
        ]
    )
    assert values == [
        True,
        True,
        True,
        "rsa_droit_ouvert_versable",
        "iae",
    ]


async def test_get_beneficiaries_without_referent(
    db_connection: Connection, snapshot: Snapshot
) -> None:
    beneficiaries_without_referent: list[
        BeneficiaryWithAdminStructureEmail
    ] = await get_beneficiaries_without_referent(db_connection)

    # Bennet n'a pas de référent et est affecté a une structure 'outdated'
    # (interlogement 93) et à la structure Groupe NS.
    # On valide donc qu'on ne notifie bien que les 2 admin de la structure Groupe NS
    assert (
        len(
            [
                beneficiary
                for beneficiary in beneficiaries_without_referent
                if beneficiary.lastname == "Bennett"
            ]
        )
        == 2
    )
    assert snapshot == beneficiaries_without_referent
