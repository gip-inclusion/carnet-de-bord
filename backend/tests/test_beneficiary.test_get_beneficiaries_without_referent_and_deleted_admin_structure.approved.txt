[BeneficiaryWithAdminStructureEmail(lastname='Gônzalez', firstname='Bolton', date_of_birth=datetime.date(1994, 9, 22), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Kerr', firstname='Bridges', date_of_birth=datetime.date(1975, 3, 3), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Patrick', firstname='Brigitte', date_of_birth=datetime.date(1961, 8, 13), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('7ea8a071-1ef8-4747-9cd8-9e68086df0e9'), name='Socio-professionnel', orientation_type=<OrientationType.sociopro: 'sociopro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Tran', firstname='Casandra', date_of_birth=datetime.date(1981, 4, 1), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Gentry', firstname='Chasity', date_of_birth=datetime.date(1972, 1, 26), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Lynch', firstname='Della', date_of_birth=datetime.date(1996, 6, 8), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Stevenson', firstname='Gilbert', date_of_birth=datetime.date(2000, 4, 17), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Roth', firstname='Hamilton', date_of_birth=datetime.date(1961, 4, 20), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Herrera', firstname='Henderson', date_of_birth=datetime.date(1969, 3, 25), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('f26ea5ba-e8a1-48ad-a031-f42d7f861008'), name='Professionnel', orientation_type=<OrientationType.pro: 'pro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Dorsey', firstname='Hendrix', date_of_birth=datetime.date(1976, 12, 18), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Solis', firstname='Hughes', date_of_birth=datetime.date(1984, 5, 3), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Aguilar', firstname='Lindsay', date_of_birth=datetime.date(1996, 6, 22), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Rice', firstname='Liz', date_of_birth=datetime.date(1982, 11, 4), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('f26ea5ba-e8a1-48ad-a031-f42d7f861008'), name='Professionnel', orientation_type=<OrientationType.pro: 'pro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Deleon', firstname='Marquez', date_of_birth=datetime.date(1979, 2, 13), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Guy', firstname='Mcclain', date_of_birth=datetime.date(1976, 8, 15), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Olsen', firstname='Mercedes', date_of_birth=datetime.date(1978, 8, 5), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Henderson', firstname='Myrna', date_of_birth=datetime.date(1999, 12, 8), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Mccormick', firstname='Pickett', date_of_birth=datetime.date(1983, 7, 1), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Nieves', firstname='Rhonda', date_of_birth=datetime.date(1988, 10, 14), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Bryant', firstname='Ross', date_of_birth=datetime.date(1960, 4, 22), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Kane', firstname='Sara', date_of_birth=datetime.date(1986, 3, 19), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Reese', firstname='Serrano', date_of_birth=datetime.date(1976, 12, 10), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Raymond', firstname='Shelly', date_of_birth=datetime.date(1989, 12, 2), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('f26ea5ba-e8a1-48ad-a031-f42d7f861008'), name='Professionnel', orientation_type=<OrientationType.pro: 'pro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Mcleod', firstname='Sherman', date_of_birth=datetime.date(1998, 10, 16), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Kline', firstname='Sonja', date_of_birth=datetime.date(1961, 12, 21), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('cc92714c-db2c-4d49-a877-571ecc6138c2'), name='Social', orientation_type=<OrientationType.social: 'social'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Malone', firstname='Steele', date_of_birth=datetime.date(2002, 6, 16), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Kaufman', firstname='Violet', date_of_birth=datetime.date(1983, 4, 20), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Walker', firstname='Woods', date_of_birth=datetime.date(1986, 4, 27), admin_structure_email='jacques.celaire@livry-gargan.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('cc92714c-db2c-4d49-a877-571ecc6138c2'), name='Social', orientation_type=<OrientationType.social: 'social'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Gônzalez', firstname='Bolton', date_of_birth=datetime.date(1994, 9, 22), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Kerr', firstname='Bridges', date_of_birth=datetime.date(1975, 3, 3), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Patrick', firstname='Brigitte', date_of_birth=datetime.date(1961, 8, 13), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('7ea8a071-1ef8-4747-9cd8-9e68086df0e9'), name='Socio-professionnel', orientation_type=<OrientationType.sociopro: 'sociopro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Tran', firstname='Casandra', date_of_birth=datetime.date(1981, 4, 1), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Gentry', firstname='Chasity', date_of_birth=datetime.date(1972, 1, 26), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Lynch', firstname='Della', date_of_birth=datetime.date(1996, 6, 8), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Stevenson', firstname='Gilbert', date_of_birth=datetime.date(2000, 4, 17), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Roth', firstname='Hamilton', date_of_birth=datetime.date(1961, 4, 20), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Herrera', firstname='Henderson', date_of_birth=datetime.date(1969, 3, 25), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('f26ea5ba-e8a1-48ad-a031-f42d7f861008'), name='Professionnel', orientation_type=<OrientationType.pro: 'pro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Dorsey', firstname='Hendrix', date_of_birth=datetime.date(1976, 12, 18), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Solis', firstname='Hughes', date_of_birth=datetime.date(1984, 5, 3), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Aguilar', firstname='Lindsay', date_of_birth=datetime.date(1996, 6, 22), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Rice', firstname='Liz', date_of_birth=datetime.date(1982, 11, 4), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('f26ea5ba-e8a1-48ad-a031-f42d7f861008'), name='Professionnel', orientation_type=<OrientationType.pro: 'pro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Deleon', firstname='Marquez', date_of_birth=datetime.date(1979, 2, 13), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Guy', firstname='Mcclain', date_of_birth=datetime.date(1976, 8, 15), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Olsen', firstname='Mercedes', date_of_birth=datetime.date(1978, 8, 5), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Henderson', firstname='Myrna', date_of_birth=datetime.date(1999, 12, 8), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Mccormick', firstname='Pickett', date_of_birth=datetime.date(1983, 7, 1), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Nieves', firstname='Rhonda', date_of_birth=datetime.date(1988, 10, 14), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Bryant', firstname='Ross', date_of_birth=datetime.date(1960, 4, 22), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Kane', firstname='Sara', date_of_birth=datetime.date(1986, 3, 19), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Reese', firstname='Serrano', date_of_birth=datetime.date(1976, 12, 10), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Raymond', firstname='Shelly', date_of_birth=datetime.date(1989, 12, 2), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('f26ea5ba-e8a1-48ad-a031-f42d7f861008'), name='Professionnel', orientation_type=<OrientationType.pro: 'pro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Mcleod', firstname='Sherman', date_of_birth=datetime.date(1998, 10, 16), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Kline', firstname='Sonja', date_of_birth=datetime.date(1961, 12, 21), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('cc92714c-db2c-4d49-a877-571ecc6138c2'), name='Social', orientation_type=<OrientationType.social: 'social'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Malone', firstname='Steele', date_of_birth=datetime.date(2002, 6, 16), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Kaufman', firstname='Violet', date_of_birth=datetime.date(1983, 4, 20), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Walker', firstname='Woods', date_of_birth=datetime.date(1986, 4, 27), admin_structure_email='lara.pafromage@cd93.fr', structure_name="Centre Communal d'action social Livry-Gargan", orientation=OrientationSystem(id=UUID('cc92714c-db2c-4d49-a877-571ecc6138c2'), name='Social', orientation_type=<OrientationType.social: 'social'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Cobb', firstname='Alexandria', date_of_birth=datetime.date(1993, 6, 30), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=OrientationSystem(id=UUID('7ea8a071-1ef8-4747-9cd8-9e68086df0e9'), name='Socio-professionnel', orientation_type=<OrientationType.sociopro: 'sociopro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Rush', firstname='Audra', date_of_birth=datetime.date(1998, 12, 22), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Mathews', firstname='Cherie', date_of_birth=datetime.date(1970, 1, 22), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Cash', firstname='Corinne', date_of_birth=datetime.date(1970, 10, 13), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Dunlap', firstname='Dominique', date_of_birth=datetime.date(1976, 3, 24), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Skinner', firstname='Edwina', date_of_birth=datetime.date(1973, 5, 14), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=OrientationSystem(id=UUID('7ea8a071-1ef8-4747-9cd8-9e68086df0e9'), name='Socio-professionnel', orientation_type=<OrientationType.sociopro: 'sociopro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Jacobson', firstname='Hester', date_of_birth=datetime.date(1962, 3, 11), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=OrientationSystem(id=UUID('f26ea5ba-e8a1-48ad-a031-f42d7f861008'), name='Professionnel', orientation_type=<OrientationType.pro: 'pro'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Townsend', firstname='Hilary', date_of_birth=datetime.date(1977, 3, 12), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Terry', firstname='Horn', date_of_birth=datetime.date(1967, 1, 18), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Conley', firstname='Isabella', date_of_birth=datetime.date(1996, 10, 27), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Chandler', firstname='Katharine', date_of_birth=datetime.date(1997, 7, 11), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Beach', firstname='Katrina', date_of_birth=datetime.date(2002, 3, 1), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Rosales', firstname='King', date_of_birth=datetime.date(1975, 1, 25), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Keller', firstname='Noel', date_of_birth=datetime.date(1977, 3, 23), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Bennett', firstname='Payne', date_of_birth=datetime.date(1963, 6, 2), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Crawford', firstname='Shaw', date_of_birth=datetime.date(1969, 11, 26), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Benjamin', firstname='Whitley', date_of_birth=datetime.date(1984, 8, 10), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Groupe NS', orientation=None),
 BeneficiaryWithAdminStructureEmail(lastname='Carlson', firstname='Oconnor', date_of_birth=datetime.date(1988, 3, 8), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Interlogement 93', orientation=OrientationSystem(id=UUID('cc92714c-db2c-4d49-a877-571ecc6138c2'), name='Social', orientation_type=<OrientationType.social: 'social'>, created_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc), updated_at=datetime.datetime(2021, 9, 21, 13, 6, 45, 76000, tzinfo=datetime.timezone.utc))),
 BeneficiaryWithAdminStructureEmail(lastname='Gallegos', firstname='Winnie', date_of_birth=datetime.date(1969, 11, 8), admin_structure_email='lara.pafromage@cd93.fr', structure_name='Interlogement 93', orientation=None)]
